### DISCUSSION TITLE
Guest users cannot use SPFX webparts, by design?

### DISCUSSION THREAD

**Discussion by McBamm** [SharePoint/sp-dev-docs#8810] [Q&A]
I've encountered an issue where External / Guest users cannot use a SPFX Webpart. Webparts instead return the error below:

```
Something went wrong If the problem persists, contact the site administrator and give them the information in Technical Details. TECHNICAL DETAILS ERROR: Script error for: fe2c2d36-2e8e-422c-a39f-79894b800ac0_0.0.1/ControlStrings http://requirejs.org/docs/errors.html#scripterror

CALL STACK: Error: Script error for: fe2c2d36-2e8e-422c-a39f-79894b800ac0_0.0.1/ControlStrings http://requirejs.org/docs/errors.html#scripterror at B (eval at e (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-gb_742e31850b13116ea900f1bc412105ec.js:70:746505), :8:252) at HTMLScriptElement.onScriptError (eval at e (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-gb_742e31850b13116ea900f1bc412105ec.js:70:746505), :30:244)
```

From my research so far it appears that the App Catalogue for a Tenancy is not shared externally by default; we've attempted to introduce "Everyone" access through PnP PowerShell, however, that's not working for the Tenancies on which we've seen this issue.

So I guess my questions are:

Is there a replicable work-around for this issue?

If external sharing is disabled by default, what's the recommended alternative approach for the issue? Is this documented anywhere?

**Comment by AJIXuMuK** [comment]
@McBamm - could you please elaborate a bit on that?
If external sharing and ability to add guest users is disabled, they should not have access to the SharePoint.
Or are you saying that Guests users are enabled for the tenant by external sharing from SharePoint Admin Center is disabled?

**Reply by AJIXuMuK** [reply]
Another approach btw is enabling Public CDN to serve the resources. This will work for any user

**Reply by McBamm** [reply]
Scenario is that External Sharing is enabled and Guest Users can access SharePoint sites to which they have been assigned permissions.  
However, custom SPFX web-parts do not load successfully and display the error illustrated above.

Unfortunately, for the organisation involved in the issue, enabling Public CDN to serve the resources was not an option.  Therefore, after some research I was able to fix the issue by following the steps I have outlined here:

[https://sharepoint.stackexchange.com/questions/306601/guest-users-cannot-use-spfx-web-parts-by-design](https://sharepoint.stackexchange.com/questions/306601/guest-users-cannot-use-spfx-web-parts-by-design)

**Reply by AJIXuMuK** [reply]
Sorry - I thought you mention PowerShell is not an option.
But yes - adding All Users is the way to go.
I usually point to this documentation for the steps: https://www.crowcanyon.help/article/431/
